test/model/message_list_test.dart

```diff
@@ -427,6 +427,53 @@ void main() {
     });
   });
 
+  group('renarrowAndFetch', () {
+    test('smoke', () => awaitFakeAsync((async) async {
+      final channel = eg.stream();
+
+      const narrow = CombinedFeedNarrow();
+      await prepare(narrow: narrow, stream: channel);
+      final messages = List.generate(100,
+        (i) => eg.streamMessage(id: 1000 + i, stream: channel));
+      await prepareMessages(foundOldest: false, messages: messages);
+
+      // …do something unrelated…
+      connection.prepare(json: olderResult(
+        anchor: 1000, foundOldest: false,
+        messages: List.generate(100,
+          (i) => eg.streamMessage(id: 900 + i, stream: channel)),
+      ).toJson());
+      await model.fetchOlder();
+      checkNotified(count: 2);
```

Comment:
What's the role of this step in the narrative of the test? It feels like it'd all work the same way without this.